src/Multivariate/ProductFun.jl

```diff
@@ -5,6 +5,15 @@
 
 export ProductFun
 
+## TODO:
+## In a newer version, an abstract type of ProducFun is needed, where different implementations are possible
+## however, refactoring this is a lot of effort...
+struct TensorIteratorFun{d, SS<:TensorSpace{<:NTuple{d, <:UnivariateSpace}}, T<:Number} <: MultivariateFun{T, d}
+    space::SS
+    coefficients::Vector{T} 
+    iterator::TrivialTensorizer{d}
+    orders::Block
```

Comment:
`Block` is an abstract type, this should ideally be concrete. Making this concrete will likely improve performance.

Reply:
fixed